### PR TITLE
Route rule rename2

### PR DESCRIPTION
**Pull request by GregHanson**
vendor PR https://github.com/istio/vendor-istio/pull/21

### CODE REVIEW

**Comment by GregHanson** [comment]
follow up to abandoned PR:
https://github.com/istio/istio/pull/3843

**Comment by ldemailly** [comment]
/lgtm

thanks so much for doing this (sorry the linearization of vendor meant you had to do it a few times)

**Comment by ldemailly** [comment]
/lgtm

curious about the prow failure but lets see if it was a flake

File: install/kubernetes/templates/istio-pilot.yaml.tmpl [hunk 1]
```diff
@@ -47,14 +47,14 @@ spec:
...
 spec:
   group: config.istio.io
   names:
-    kind: V1alpha2RouteRule
-    listKind: V1alpha2RouteRuleList
-    plural: v1alpha2routerules
-    singular: v1alpha2routerule
+    kind: VirtualService
+    listKind: VirtualServiceList
+    plural: sirtualservices
```
Comment (ldemailly) [inline install/kubernetes/templates/istio-pilot.yaml.tmpl]: ^^^ beep beep

**Comment by ldemailly** [comment]
resolving conflict in #3887

**Comment by ldemailly** [comment]
merged as  #3887 + follow up fun #3890

**Pull request closed without merging**